### PR TITLE
Implement Puppet and Ansible property override.

### PR DESCRIPTION
**Pull request by nelsonjr**

### CODE REVIEW

File: provider/ansible/property_override.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2017 Google Inc.
```
Comment (rosbo) [inline provider/ansible/property_override.rb]: Add in the issue title that you are also adding it for ansible.

File: api/resource.rb [hunk 2]
```diff
@@ -45,6 +44,9 @@ module Properties
 
     include Properties
 
+    # Parameters can be overridden via Provider::PropertyOverride
+    attr_reader :parameters
```
Comment (rosbo) [inline api/resource.rb]: You will need to add some code in resource_overrides to enable this.

File: provider/resource_overrides.rb [hunk 3]
```diff
@@ -100,6 +100,7 @@ def override_objects
         raise "The resource to override must exist #{name}" if api_object.nil?
         check_property_value 'overrides', override, Provider::ResourceOverride
         override.apply api_object
+        populate_nonoverridden_properties api_object, override
```
Comment (rosbo) [inline provider/resource_overrides.rb]: Please rebase.

File: api/resource.rb (hunk 2 above)
Comment (nelsonjr) [reply api/resource.rb]: I did already. Note that in `get_properties` I changed it to be `api_entity.all_user_properties`, which will include all parameters as well.

File: provider/resource_overrides.rb (hunk 3 above)
Comment (nelsonjr) [reply provider/resource_overrides.rb]: It is rebased.

File: provider/ansible/property_override.rb (hunk 1 above)
Comment (nelsonjr) [reply provider/ansible/property_override.rb]: Done.

File: api/resource.rb (hunk 2 above)
Comment (rosbo) [reply api/resource.rb]: Gotcha 👍

**Pull request merged**